### PR TITLE
Use structured clone-like algorithm for Hybrid's result and arguments serialization (#524)

### PR DESCRIPTION
**Pull request by inikulin**
\cc @AlexanderMoskovkin @helen-dikareva

### CODE REVIEW

**Comment by inikulin** [comment]
ping @AlexanderMoskovkin @helen-dikareva

**Comment by helen-dikareva** [comment]
lgtm

**Comment by AlexanderMoskovkin** [comment]
lgtm
/cc @VasilyStrelyaev

File: src/test-run/commands/replicator.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+// WARNING: this file is used by both the client and the server.
+// Do not use any browser or node-specific API!
+// -------------------------------------------------------------
+
+import Replicator from 'replicator';
+
+var identityFn = val => val;
+
+// NOTE: we will serialize replicator results
+// to JSON with command or command result.
```
Comment (VasilyStrelyaev) [inline src/test-run/commands/replicator.js]:
```
// to JSON with a command or command result.
```

File: src/test-run/commands/index.js [hunk 2]
```diff
@@ -464,63 +471,76 @@ export class TestDoneCommand {
 
 // Factory
 export function createCommandFromObject (obj) {
-    if (obj.type === TYPE.click)
-        return new ClickCommand(obj);
+    /* eslint-disable indent*/
+    // TODO: eslint raises an 'incorrect indent' error here. We use
+    // the old eslint version (v1.x.x). We should migrate to v2.x.x
```
Comment (VasilyStrelyaev) [inline src/test-run/commands/index.js]:
```
    // an old eslint version (v1.x.x). We should migrate to v2.x.x
```

File: src/test-run/commands/index.js [hunk 3]
```diff
@@ -464,63 +471,76 @@ export class TestDoneCommand {
...
+            return new TestDoneCommand();
+    }
+    /* eslint-enable indent*/
 }
 
 export function isCommandRejectableByPageError (command) {
-    return !isTestDoneCommand(command) && command.type !== TYPE.execHybridFn && command.type !== TYPE.waitForElement;
+    /* eslint-disable indent*/
+    // TODO: eslint raises an 'incorrect indent' error here. We use
+    // the old eslint version (v1.x.x). We should migrate to v2.x.x
```
Comment (VasilyStrelyaev) [inline src/test-run/commands/index.js]:
```
    // an old eslint version (v1.x.x). We should migrate to v2.x.x
```

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by VasilyStrelyaev** [comment]
(not sure there was a need to correct TODOs, but... there's nothing more permanent than temporary)

**Comment by inikulin** [comment]
FPR

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by AlexanderMoskovkin** [comment]
lgtm
I'll merge when functional tests are finished

**Pull request merged**